### ISSUE TITLE
报错 uncaught  (in promise) undefined

### ISSUE THREAD

**Issue by waney** [vueComponent/ant-design-vue-pro#391]
**Describe the bug （描述 Bug）**

vue-router版本 > 3.0.7 报错如下：
`uncaught  (in promise) undefined`

![微信截图_20190823163503](https://user-images.githubusercontent.com/69211/63578902-07678f00-c5c4-11e9-82c1-e674426dc208.png)

> 更换版本解决
`yarn upgrade vue-router@3.0.7`

https://github.com/vuejs/vue-router/blob/dev/CHANGELOG.md

**3.1.2 (2019-08-08)**
Bug Fixes
types: prioritize promise based push/replace (1243e8b)
Reverts
"fix(hash): correctly place query if placed before hash (#2851)" (9b30e4c), closes #2876. See more information at https://github.com/vuejs/vue-router/issues/2125#issuecomment-519521424

**3.1.1 (2019-08-06)**
Bug Fixes
link: silence back navigations errors (59b6da3)

**3.1.0 (2019-08-06)**
Bug Fixes
abstract history: allow router.back in abstract mode when 2 consecutive same routes appear in history stack (#2771) (8910979), closes #2607
hash: correctly place query if placed before hash (#2851) (b7715dc), closes #2125 #2262
link: Fix active links when parent link redirects to child (#2772) (64785a9), closes #2724
adapt error to work on IE9 (527d6d5)
Features
alias: warn against redundant aliases (04a02c0), closes #2461 #2462
scroll: handle id selectors starting with a number (799ceca), closes #2163
return a promise with push and replace (#2862) (d907a13), closes #1769 #1769
scoped slot for link (e289dde)
warn the user for invalid uses of v-slot with Link (44c63a9)

**3.0.7 (2019-07-03)**
Bug Fixes
apps loaded from Windows file shares not mapped to network drives (#2774) (c2c78a3)
make callback of next in beforeRouterEnter more consistent (#2738) (8ac478f), closes #2761 #2728

**Comment by sendya** [comment]
在什么情况下 **报错**

**Comment by waney** [comment]
> 在什么情况下 **报错**

@sendya 

在 vue-router 版本 > 3.0.7 的时候。
从登录页进入的时候，根据权限生成路由，会报此错误。后续使用还是正常。

**Comment by waney** [comment]
不算BUG哈，只是发现顺手贴过来。vue-router 8月发了好几个版本。

**Comment by sendya** [comment]
好的，我可以在近期升级一下最低版本

**Comment by waney** [comment]
问题出在登录成功后触发跳转时：
/src/views/user/Login.vue
`this.$router.push({ name: 'dashboard' })`
此时权限后生成的路由还没完成。

然后Vue Router 这样说 
https://router.vuejs.org/zh/guide/essentials/navigation.html
router.push(location, onComplete?, onAbort?)

> onComplete?, onAbort?


```loginSuccess () {
      this.isLoginError = false
      this.$router.push('/', () => {
        this.$notification.success({
          message: '欢迎',
          description: `${timeFix()}，欢迎回来`
        })
      })
```

**Comment by sendya** [comment]
fixed 3dee851b9718b0b2b934566fdc4a2acbaded1b49

**Comment by Oralinge** [comment]
你好，请问这个有解决的方法吗？
